### PR TITLE
﻿CU-86drua2h0 - NEON3 - Wallets Screen - Update Application to use new contextual menu (side bar) + Overview Tab

### PR DESCRIPTION
**Pull request by Thacryba**
**What current issue(s) from Trello/Github does this address?**

**What problem does this PR solve?**

**How did you solve this problem?**

**How did you make sure your solution works?**

**Are there any special changes in the code that we should be aware of?**

**Is there anything else we should know?**

- [ ] Unit tests written?

### CODE REVIEW

**Comment by melanke** [comment]
Task linked: [CU-86drua2h0 NEON3 - Wallets Screen - Update Application to use new contextual menu (side bar) + Overview Tab](https://app.clickup.com/t/86drua2h0)

File: src/renderer/src/helpers/UtilsHelper.ts [hunk 1]
```diff
@@ -64,6 +64,14 @@ export class UtilsHelper {
     navigator.clipboard.writeText(text)
   }
 
+  static truncateAddress(text?: string) {
+    if (!text || text.length < 8) return ''
+    const textLenght = text.length
+    const firstPart = text.slice(0, 4)
+    const lastPart = text.slice(textLenght - 4, textLenght)
+    return `${firstPart}...${lastPart}`
+  }
```
Comment (raulduartep) [inline src/renderer/src/helpers/UtilsHelper.ts]: You don't need this new method, you can use `StringHelper.truncateStringMiddle` instead

File: src/renderer/src/routes/pages/Wallets/AccountSidebarLink.tsx
Comment (raulduartep) [inline src/renderer/src/routes/pages/Wallets/AccountSidebarLink.tsx:1]: Use the PortfolioSidebarLink component. Rebase with dev-neon3 to get this component. You can move it to components folder and rename it to a more generic name.

File: src/renderer/src/routes/pages/Wallets/AccountOverview/index.tsx [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+import { useAccountsSelector } from '@renderer/hooks/useAccountSelector'
+import { useBalancesAndExchange } from '@renderer/hooks/useBalancesAndExchange'
+
+export const AccountOverview = () => {
+  const { t } = useTranslation('pages', { keyPrefix: 'accountOverview' })
+  const { t: commonGeneral } = useTranslation('common', { keyPrefix: 'general' })
+  const navigate = useNavigate()
+  const { address } = useParams()
+  const { accounts } = useAccountsSelector()
+  const [selectedAccount, setSelectedAccount] = useState<IAccountState>()
```
Comment (raulduartep) [inline src/renderer/src/routes/pages/Wallets/AccountOverview/index.tsx]: You can use a useMemo here instead.
```suggestion
    const account = useMemo(() => accounts.find(account => account.address === address), [accounts, address])

```

File: src/renderer/src/routes/pages/Wallets/AccountOverview/index.tsx [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+
+  useEffect(() => {
+    const account = accounts.find(account => account.address === address)
+    setSelectedAccount(account)
+  }, [accounts, address])
```
Comment (raulduartep) [inline src/renderer/src/routes/pages/Wallets/AccountOverview/index.tsx]: If you use useMemo, you won't need this

File: src/renderer/src/locales/en/common.json [hunk 4]
```diff
@@ -6,7 +6,9 @@
     "edit": "Edit",
     "cancel": "Cancel",
     "reset": "Reset",
-    "successfullyCopied": "Successfully copied to clipboard"
+    "successfullyCopied": "Successfully copied to clipboard",
+    "send": "Send",
+    "receive": "Receive"
```
Comment (raulduartep) [inline src/renderer/src/locales/en/common.json]: Move this to translation of the component that is using it

File: src/renderer/src/locales/en/pages.json [hunk 5]
```diff
@@ -164,5 +170,10 @@
       "title": "Select a receiving account",
       "selectReceivingAccount": "Select receiving account"
     }
+  },
+  "accountOverview": {
+    "accountOverview": "Account overview",
+    "holdings": "Holdings",
+    "balance": "Balance"
```
Comment (raulduartep) [inline src/renderer/src/locales/en/pages.json]: Move this to wallet page translation

**Pull request merged**